### PR TITLE
Simplify loading columns into cache in CachingJdbcClient

### PR DESCRIPTION
**Pull request by zaz968m**
Utilize guava cache.get(key, loader) instead of custom way of loading.
loader is able to handle external context, which addresses constraint:
columns depend on session and session cannot be used in cache key

This pr is a small step toward merging https://github.com/prestosql/presto/pull/6067

### CODE REVIEW

**Review by kokosing — APPROVED** [review]

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java [hunk 1]
```diff
@@ -106,15 +106,8 @@ public boolean schemaExists(JdbcIdentity identity, String schema)
...
-
         ColumnsCacheKey key = new ColumnsCacheKey(JdbcIdentity.from(session), tableHandle.getSchemaTableName());
-        List<JdbcColumnHandle> columns = columnsCache.getIfPresent(key);
-        if (columns != null) {
-            return columns;
-        }
-        columns = delegate.getColumns(session, tableHandle);
-        columnsCache.put(key, columns);
-        return columns;
+        return get(columnsCache, key, () -> delegate.getColumns(session, tableHandle));
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java:110]: > Use built-in thread safe cache loading for columns

The commit may be read as if the previous code was not thread safe, while i think it was.
am i missing something?

if it is just a refactor, then perhaps name the commit as such, eg
```
Simplify loading columns into cache in CachingJdbcClient

Utilize `cache.get` with a loader, to simplify control flow.
```

**Review by skrzypo987 — APPROVED** [review]

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java (hunk 1 above)
Comment (zaz968m) [reply presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java:110]: I understand it in this way:
1. `getColumns` method does not have any synchronisation so two concurrent clients can invoke it. 
2. `columnsCache.getIfPresent(key)` returns `false` for both clients.
3. `delegate.getColumns(session, tableHandle)` is invoked for each of the clients 

In any case I can update message.

Comment (kokosing) [reply presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java:110]: > the previous code was not thread safe

It was thread safe, but not optimal. There was a case where we could load the same cache entry concurrently by several threads. Not it should be fixed.

@zaz968m Can you please add a test for that?

Comment (findepi) [reply presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java:110]: > It was thread safe, but not optimal. There was a case where we could load the same cache entry concurrently by several threads. Not it should be fixed.

this is not a _problem_.

actually, the new version could be seen as introducing unnecessary contention/synchronization. I do not think it is an actual problem either, but i do insist we are not fixing any problem other than code readability and so this should be reflected in cmt msg appropriately.

Comment (zaz968m) [reply presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java:110]: @kokosing as agreed, test will be delivered in separate pr.

Comment (zaz968m) [reply presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java:110]: @findepi commit message is updated. Please AC.

**Review by findepi — APPROVED** [review]

**Pull request merged**